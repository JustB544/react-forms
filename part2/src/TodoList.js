import React, {useState} from "react";
import Todo from "./Todo";
import {v4 as uuid} from 'uuid';
import NewTodoForm from "./NewTodoForm";

function TodoList(props) {
    const [todos, setTodos] = useState(props.initial || {});

    function deleteTodo(id){
        setTodos(({[id]: undefined, ...t}) => t);
    }

    function createTodo({ task }){
        setTodos(t => ({...t, [uuid()]: { task }}));
    }

    return (
        <div>
            { Object.keys(todos).map(t => <Todo 
            key={t} 
            id={t} 
            task={todos[t].task}
            deleteTodo={evt => deleteTodo(t)}/>) }
            <NewTodoForm createTodo={(todo, evt) => createTodo(todo)}/>
        </div>
    );
}

export default TodoList;