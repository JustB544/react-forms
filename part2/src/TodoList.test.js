import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";
import TEST_TODOS from "./_testcommon";

it("renders", () => {
    render(<TodoList />);
});
  
it("matches snapshot", () => {
    const {asFragment} = render(<TodoList initial={TEST_TODOS}/>);
    expect(asFragment()).toMatchSnapshot();
});

it("deletes a todo", () => {
    const {container} = render(<TodoList initial={TEST_TODOS}/>);
    const todo1 = container.querySelector("#todo1 button");
    fireEvent.click(todo1);
    expect(todo1).not.toBeInTheDocument();
});