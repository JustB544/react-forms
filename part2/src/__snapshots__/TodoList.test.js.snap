// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="Todo"
      id="todo1"
    >
      <h3>
        Complete homework
      </h3>
      <button>
        X
      </button>
    </div>
    <div
      class="Todo"
      id="todo2"
    >
      <h3>
        New task
      </h3>
      <button>
        X
      </button>
    </div>
    <form
      style="display: flex; flex-direction: column;"
    >
      <div>
        <label
          for="task"
        >
          Task: 
        </label>
        <input
          name="task"
          placeholder="New task"
          type="text"
          value=""
        />
      </div>
      <button>
        Create
      </button>
    </form>
  </div>
</DocumentFragment>
`;
